In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats

# 1. A physician is evaluating a new diet for her patients with a family history of heart disease. To test the effectiveness of this diet, 16 patients are placed on the diet for 6 months. Their weights and triglyceride levels are measured before and after the study, and the physician wants to know if either set of measurements has changed.

In [2]:
diet_study =  pd.read_csv('C:/python_casestudies/Basic Stats Case Study_1/dietstudy.csv')
diet_study.head()

patid  age  gender  tg0  tg1  tg2  tg3  tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0      1   45    Male  180  148  106  113  100   198   196   193   188   192
1      2   56    Male  139   94  119   75   92   237   233   232   228   225
2      3   50    Male  152  185   86  149  118   233   231   229   228   226
3      4   46  Female  112  145  136  149   82   179   181   177   174   172
4      5   64    Male  156  104  157   79   97   219   217   215   213   214

#H0 : There is no change in the their measurments of weights and triglyceride levels before and after the study.
#Ha : There is a change in the their measurments of weights and triglyceride levels before and after the study.


In [3]:
# CI = 95% | p-value = 0.05
#print the average for both weights and triglyceride levels before the 6 months of the diet plan and post 6 months.

#compare the average for weights of before and after diet plan.
print('mean_wgt0 :', diet_study.wgt0.mean(), '| Mean_wgt4 :', diet_study.wgt4.mean())

mean_wgt0 : 198.375 | Mean_wgt4 : 190.3125


In [4]:
#compare the average for triglyceride levels of before and after diet plan.
print('mean_tg0 :', diet_study.tg0.mean(), '| Mean_tg4 :', diet_study.tg4.mean())

mean_tg0 : 138.4375 | Mean_tg4 : 124.375


In case of (Triglyceride & Final Triglyceride), the difference in means of the two sample sets(triglyceride levels after the diet period of 6 months) is 14.06. Therefore, across all 16 patients triglyceride levels dropped on average after 6 months of new diet 
#2. In case of (Weight & Final Weight), the difference in means of the two sample sets(weights after the diet period of 6 months) is 8.062. Therefore, across all 16 patients weights dropped on average after 6 months of new diet about 8 pounds.

In [5]:
stats.ttest_rel(diet_study.wgt0, diet_study.wgt4)

Ttest_relResult(statistic=11.174521688532522, pvalue=1.137689414996614e-08)

#perform the ttest for weights before the diet and after the diet plan.

stage 2

H0: levels of Triglycerides of individual before diet == levels of Triglycerides of individual after diet

H1: levels of Triglycerides of individual before diet != levels of Triglycerides of individual after diet

In [6]:
stats.ttest_rel(diet_study.tg0, diet_study.tg4)

Ttest_relResult(statistic=1.2000008533342437, pvalue=0.24874946576903698)

# conlusion : 
1. Since the p-value '1.137689414996614e-08' for change in weight is less than 0.05,we can conclude that the average loss of 8.06 pounds per patient is not due to chance variation, and can be attributed to the diet.

2. However, the p-value '0.24874946576903698' greater than 0.05 for change in triglyceride level shows the diet did not significantly reduce their triglyceride levels.

# 2. An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales?

In [7]:
credit_promo = pd.read_csv("C:/python_casestudies/Basic Stats Case Study_1/creditpromo.csv")
credit_promo.head(10)

id         insert      dollars
0   148       Standard  2232.771979
1   572  New Promotion  1403.807542
2   973       Standard  2327.092181
3  1096       Standard  1280.030541
4  1541  New Promotion  1513.563200
5  1947  New Promotion  1729.627996
6  2001  New Promotion  1609.705918
7  2130       Standard  1476.624884
8  2616       Standard  1460.769753
9  2886  New Promotion  1854.489028

In [8]:
std = credit_promo.loc[(credit_promo['insert'] == 'Standard'), 'dollars']
promo = credit_promo.loc[(credit_promo['insert'] == 'New Promotion'), 'dollars']

print('mean_std :', std.mean(), '| Mean_promo :', promo.mean())

mean_std : 1566.3890309659348 | Mean_promo : 1637.4999830647992


. On average, customers who received the interest-rate promotion charged about $70 more than the normal standard season ad, and they vary a little more around their average.

#H0 : sales of seasonal ad = sales of promotional ad.

#Ha : sales of seasonal ad <> sales of promotional ad.

#CI = 95% | p-value = 0.05

In [9]:
stats.ttest_ind(std, promo)

Ttest_indResult(statistic=-2.2604227264649963, pvalue=0.024225996894147814)

# Conclusion :
For the above test, the p-value is 0.024225996894147814 is less than 0.05. so, We reject null hypothesis.

Since the significance value of the test is less than 0.05, we can safely conclude that the average of 71.11 dollars more spent by cardholders receiving the reduced interest rate is not due to chance alone. 
The store will now consider extending the offer to all credit customers.

# 3. An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). 

In [10]:
pollination = pd.read_csv("C:/python_casestudies/Basic Stats Case Study_1/pollination.csv")
pollination.head()

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0  Natural      1.85            147.70            16.86
1  Natural      1.86            136.86            16.77
2  Natural      1.83            149.97            16.35
3  Natural      1.89            172.33            18.26
4  Natural      1.80            144.46            17.90

a. Is the overall population of Seed yield/plant (g) equals to 200?

In [11]:
pollination.Seed_Yield_Plant.mean()

180.8035

#H0 : population = 200
#Ha : population < 200
#CI : 95% | p-value 0.05


In [12]:
# perform the test
stats.ttest_1samp(pollination.Seed_Yield_Plant, 200)

Ttest_1sampResult(statistic=-2.3009121248548645, pvalue=0.032891040921283025)

Conclusion :

Since the p-value '0.032891040921283025' of the test is less than 0.05, we reject the null hypothesis. 
Therefore overall population of seed yield/plant is not equal to 200.

# b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different.

In [13]:
natural_fruit_wt = pollination.loc[pollination['Group'] == 'Natural', 'Fruit_Wt']
hand_fruit_wt = pollination.loc[pollination['Group'] == 'Hand', 'Fruit_Wt'] 

#perform the Anova.
stats.f_oneway(natural_fruit_wt,hand_fruit_wt)

F_onewayResult(statistic=312.228532974426, pvalue=8.078362076486568e-13)

In [14]:
natural_Seed_Yield_Plant = pollination.loc[pollination['Group'] == 'Natural', 'Seed_Yield_Plant']
hand_Seed_Yield_Plant = pollination.loc[pollination['Group'] == 'Hand', 'Seed_Yield_Plant'] 

#perform the Anova.
stats.f_oneway(natural_Seed_Yield_Plant,hand_Seed_Yield_Plant)

F_onewayResult(statistic=194.83303662980398, pvalue=4.271481585484407e-11)

In [15]:
natural_Seedling_length = pollination.loc[pollination['Group'] == 'Natural', 'Seedling_length']
hand_Seedling_length = pollination.loc[pollination['Group'] == 'Hand', 'Seedling_length'] 

#perform the test.
stats.f_oneway(natural_Seedling_length,hand_Seedling_length)

F_onewayResult(statistic=6.46293337115627, pvalue=0.020428817064110556)

# 4. An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently?

In [16]:
ele_firm = pd.read_csv("C:/python_casestudies/Basic Stats Case Study_1/dvdplayer.csv")
ele_firm.head()

agegroup   dvdscore
0  65 and over  38.454803
1        55-64  17.669677
2  65 and over  31.704307
3  65 and over  25.924460
4     Under 25  30.450007

In [17]:
#H0 : customers of different age group rated dvd similarly
#Ha : customers of different age froup rated dvd score differently
#CI : 95% | p-value = 0.05

In [18]:
age_group_1 = ele_firm['dvdscore'].loc[ele_firm['agegroup']=="65 and over"]
age_group_2 = ele_firm['dvdscore'].loc[ele_firm['agegroup']=="55-64"]
age_group_3 = ele_firm['dvdscore'].loc[ele_firm['agegroup']=="Under 25"]
age_group_4 = ele_firm['dvdscore'].loc[ele_firm['agegroup']=="35-44"]
age_group_5 = ele_firm['dvdscore'].loc[ele_firm['agegroup']=="45-54"]
age_group_6 = ele_firm['dvdscore'].loc[ele_firm['agegroup']=="25-34"]

#perform the Anova.

stats.f_oneway(age_group_1, age_group_2, age_group_3, age_group_4, age_group_5, age_group_6)

F_onewayResult(statistic=6.992526962676517, pvalue=3.087324905679639e-05)

# Conclusion :

The p value is '3.087324905679639e-05' is less than 0.05. so, We reject null hypothesis

Therefore,consumers of various ages rated the design differently

# 5. A survey was conducted among 2800 customers on several demographic characteristics. Working status, sex, age, age-group, race, happiness, no. of child, marital status, educational qualifications, income group etc. had been captured for that purpose.

In [20]:
cust_survey = pd.read_csv("C:/python_casestudies/Basic Stats Case Study_1/sample_survey.csv")
cust_survey.head()

id            wrkstat        marital  childs   age  educ  paeduc  maeduc  \
0   1  Working full time       Divorced     2.0  60.0  12.0    12.0    12.0   
1   2  Working part-time  Never married     0.0  27.0  17.0    20.0     NaN   
2   3  Working full time        Married     2.0  36.0  12.0    12.0    12.0   
3   4  Working full time  Never married     0.0  21.0  13.0     NaN    12.0   
4   5  Working full time  Never married     0.0  35.0  16.0     NaN    12.0   

   speduc          degree  ...        agecat childcat news1 news2  news3  \
0     NaN     High school  ...      55 to 64      1-2    No    No     No   
1     NaN  Junior college  ...      25 to 34     None    No    No    Yes   
2    16.0     High school  ...      35 to 44      1-2    No    No     No   
3     NaN     High school  ...  Less than 25     None    No    No     No   
4     NaN        Bachelor  ...      35 to 44     None    No    No     No   

  news4 news5      car1      car2      car3  
0    No    No  American  Japanese  Japanese  
1    No    No  American    German  Japanese  
2   Yes   Yes  American  American       NaN  
3   Yes   Yes  American     Other       NaN  
4    No    No  American  American    Korean  

[5 rows x 46 columns]

# a. Is there any relationship in between labour force status with marital status?

In [21]:
#H0 : No Relationship b/w labour force status with marital status.
#Ha : There is a relation b/w labour force with marital status
#CI = 95% | p-value = 0.05

In [23]:
#get the cross table of work status and marital status.
obs = pd.crosstab(cust_survey.wrkstat, cust_survey.marital)
obs

marital                  Divorced  Married  Never married  Separated  Widowed
wrkstat                                                                      
Keeping house                  25      200             35         13       55
Other                          12       16             14          4        8
Retired                        53      168             17          6      150
School                          7        9             60          2        1
Temporarily not working         9       23             11          1        2
Unemployed, laid off           10       13             32          0        3
Working full time             295      778            392         58       44
Working part-time              35      138            102          9       20

In [24]:
#perform the chi2 test.
stats.chi2_contingency(obs)

(729.2421426572284,
 1.4875268409067568e-135,
 28,
 array([[ 51.69187279, 155.8869258 ,  76.84240283,  10.77879859,
          32.8       ],
        [  8.51024735,  25.66431095,  12.65088339,   1.7745583 ,
           5.4       ],
        [ 62.09328622, 187.25441696,  92.30459364,  12.94770318,
          39.4       ],
        [ 12.45017668,  37.5459364 ,  18.50777385,   2.59611307,
           7.9       ],
        [  7.24946996,  21.86219081,  10.77667845,   1.51166078,
           4.6       ],
        [  9.14063604,  27.56537102,  13.58798587,   1.90600707,
           5.8       ],
        [246.95477032, 744.74028269, 367.10989399,  51.495053  ,
         156.7       ],
        [ 47.90954064, 144.48056537,  71.21978799,   9.99010601,
          30.4       ]]))

# Conclusion :
p-value 1.4875268409067568e-135 is significantly less than 0.05 S0, we reject null hypothesis.

There is a relation b/w labour force with marital status. both are dependent each other.

# b. Do you think educational qualification is somehow controlling the marital status? 

In [25]:
#H0 : No relationship b/w educational qualification with marital status
#Ha : there is a relationship b/w educational qualification with marital status
# CI = 95% | p-value = 0.05

In [26]:
#get the cross table of degree and marital status.
obs = pd.crosstab(cust_survey.degree, cust_survey.marital)
obs

marital         Divorced  Married  Never married  Separated  Widowed
degree                                                              
Bachelor              58      251            129         12       28
Graduate              29      123             41          3        9
High school          241      686            367         58      148
Junior college        45      108             46          3        6
LT High school        70      174             77         17       92

In [27]:
#perform the chi2 test.
stats.chi2_contingency(obs)

(122.68449020508541,
 1.6707923432360119e-18,
 16,
 array([[ 75.06345268, 227.39312301, 111.83268345,  15.75824176,
          47.95249911],
        [ 32.19248493,  97.52215526,  47.9617157 ,   6.75824176,
          20.56540234],
        [235.55476781, 713.57674583, 350.9393832 ,  49.45054945,
         150.4785537 ],
        [ 32.66359447,  98.94930876,  48.66359447,   6.85714286,
          20.86635945],
        [ 67.52570011, 204.55866714, 100.60262318,  14.17582418,
          43.1371854 ]]))

# Conclusion :
p-value 1.6707923432360119e-18 is significantly less than 0.05 S0, we reject null hypothesis.

I think educational qualification is somehow controlling the marital status.

# c. Is happiness is driven by earnings or marital status?

In [28]:
#H0 : happiness is not driven by earnings.
#Ha : happiness is driven by earnings.
#CI = 95% | p-value = 0.05


In [43]:
#get the cross table of happy and earning.
obs = pd.crosstab(cust_survey.happy, cust_survey.income)
obs

income         $1000 TO 2999  $10000 - 14999  $15000 - 19999  $20000 - 24999  \
happy                                                                          
Not too happy              7              39              33              40   
Pretty happy              20             107             119             155   
Very happy                 5              44              26              50   

income         $25000 or more  $3000 TO 3999  $4000 TO 4999  $5000 TO 5999  \
happy                                                                        
Not too happy             113              9              9              6   
Pretty happy              888             11             13             18   
Very happy                571              4             10             11   

income         $6000 TO 6999  $7000 TO 7999  $8000 TO 9999  LT $1000  
happy                                                                 
Not too happy             14             12              9        11  
Pretty happy              13             21             30        13  
Very happy                 6             14             19        11

In [44]:
#perform the chi2 test.
stats.chi2_contingency(obs)

(178.9505306121643,
 1.4107677273473057e-26,
 22,
 array([[  3.89520355,  23.12777106,  21.66706973,  29.82265216,
         191.35187424,   2.92140266,   3.89520355,   4.26037888,
           4.01692866,   5.72108021,   7.06005643,   4.26037888],
        [ 18.16041919, 107.82748892, 101.01733172, 139.04070939,
         892.1305925 ,  13.62031439,  18.16041919,  19.86295848,
          18.72793229,  26.67311568,  32.91575977,  19.86295848],
        [  9.94437727,  59.04474002,  55.31559855,  76.13663845,
         488.51753325,   7.45828295,   9.94437727,  10.87666264,
          10.25513906,  14.60580411,  18.0241838 ,  10.87666264]]))

In [31]:
# Happiness with respect to marital status.

#H0 : happiness is not driven by marital status.
#Ha : happiness is driven by marital status.
#CI = 95% | p-value = 0.05

In [32]:
#get the cross table of happy and marital status.
obs = pd.crosstab(cust_survey.happy, cust_survey.marital)
obs

marital        Divorced  Married  Never married  Separated  Widowed
happy                                                              
Not too happy        72       71            108         30       59
Pretty happy        278      684            426         49      137
Very happy           93      582            120         13       83

In [33]:
#perform the chi2 test.
stats.chi2_contingency(obs)

(260.6894389418282,
 9.3147261197964e-52,
 8,
 array([[ 53.6969697 , 162.06060606,  79.27272727,  11.15151515,
          33.81818182],
        [248.58538324, 750.24527629, 366.98609626,  51.62495544,
         156.55828877],
        [140.71764706, 424.69411765, 207.74117647,  29.22352941,
          88.62352941]]))

# conlusion :
the p value is 9.3147261197964e-52 is less than 0.05. So, we reject null hypothesis. 

As it can be clearly seen that chi_square score for (happiness w.r.t to marital status) i.e. 260.68 is more than chi_square score for (happiness w.r.t to marital earnings) i.e 178.95, 
therefore we can conclude that happiness is driven by marital status